nexusparser/parser.py

```diff
@@ -237,6 +249,29 @@ def parse(self, mainfile: str, archive: EntryArchive, logger=None, child_archive
         self.archive.m_create(nexus.Nexus)  # type: ignore[attr-defined] # pylint: disable=no-member
         self.nxroot = self.archive.nexus
 
+        extension = pathlib.Path(mainfile).suffix
+        if extension in (".yaml", ".yml"):
+            base_dir = os.path.dirname(mainfile)
+            from nexusparser.tools.dataconverter.convert import convert, parse_params_file
+            with open(mainfile) as file:
+                conv_params = parse_params_file(file)
+
+                def check_path(path: str):
+                    """Return true if path supplied by the user is not absolute or has a ../"""
+                    if os.path.isabs(path) or ".." in path:
+                        raise Exception("The user provided an invalid path in the parameter YAML.")
+                    return path
+
+                if isinstance(conv_params["input_file"], list):
+                    conv_params["input_file"] = [f"{base_dir}{os.sep}{check_path(file)}"
+                                                 for file in conv_params["input_file"]]
+                else:
+                    conv_params["input_file"] = (f"{base_dir}{os.sep}"
+                                                 f"{check_path(conv_params['input_file'])}")
+                conv_params["output"] = f"{base_dir}{os.sep}{check_path(conv_params['output'])}"
+                convert(**conv_params)
+                mainfile = conv_params["output"]
+
         nexus_helper = read_nexus.HandleNexus(logger, [mainfile])
```

Comment:
Won't NOMAD try to parse the freshly generated new NeXus file anyway?

Reply:
The Nomad walk over the files doesn't win the race here. The user can press reprocess to have another entry for the NXS file. It's hard to enforce this and can lead to unexpected behavior.

I've went with this additional feature in h5web in Nomad that allows us to show the Nexus file assosciated with the dataconverter parameter file. Here is the relevant MR: https://gitlab.mpcdf.mpg.de/nomad-lab/nomad-FAIR/-/merge_requests/734